plugins/net_plugin/net_plugin.cpp

```diff
@@ -1188,7 +1188,7 @@ namespace eosio {
             pending_message_buffer.peek(&b, 1, index);
             which |= uint32_t(uint8_t(b) & 0x7f) << by;
             by += 7;
-         } while( uint8_t(b) & 0x80 );
+         } while( uint8_t(b) & 0x80 && by < 32);
```

Comment:
Can this code be replaced with a call to `fc::io::raw::unpack(Stream&, unsigend int&)` ?

Reply:
Sorry, no it can't. This is a peek, not a read.